### PR TITLE
W22 5pm 1 basic course search

### LINKED ISSUE #15: Basic Search Page Frontend
# Description
Create the Frontend page for a basic search. This basic search will have 3 dropdown menus, one for the quarter and one for the course level (undergraduate - lower division, undergraduate - upper division, graduate, all) and subjects, etc.

# Details
1. Create search form.
2. Create create search page 

# Acceptance Criteria

- [x] There is a Basic Search Form that has 3 dropdown menus.
- [x] There is a Basic Search page that displays the form
- [x] A toast pops up after submitting the form that says something like "correct request made to frontend"
- [x] Subject dropdown currently does not call the backend
- [x] 100% coverage test
- [x] 100% mutation testing

### PR DESCRIPTION
**Pull request by sidd328**
# Overview

The following PR creates the Basic Course Search Page frontend. Furthermore, the page has the 3 dropdown selectors (Subject selector, level selector and quarter selector)
Out of those only the level and subject selector are new (quarter selector was present from starter code)

In terms of mutation testing, we managed to test everything except the internalStorage. S21 also encountered this problem because, it is hard to exactly adjust the internalStorage (even with mocking) and test whether it works so we had to tell stryker to ignore only the internalStorage tests 

# Issues Addressed

Closes #15

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "babel-loader": "8.1.0",
     "bootstrap": "^5.0.0-beta3",
     "history": "^5.2.0",
+    "isomorphic-unfetch": "^3.1.0",
```
Comment (lukeyoffe) [inline frontend/package.json:13]: Idt this should be here

File: frontend/src/main/pages/HomePage.js [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
+
+  // const initialCourseJSON = {
+  //     "pageNumber": 1,
+  //     "pageSize": 1,
+  //     "total": 0,
+  //     "classes": []
+  // };
+
+  // // courseId, title, sectionNumber, instructor, enroll code, units, total enrolled students, max enrolled
+  // const [courseJSON, setCourseJSON] = useState(initialCourseJSON);
```
Comment (lukeyoffe) [inline frontend/src/main/pages/HomePage.js:32]: is this commented out code necessary?

**Review by lukeyoffe — COMMENTED** [review]
See comments

File: frontend/package.json (hunk 1 above)
Comment (asharma381) [reply frontend/package.json:13]: yes this shouldn't be here

File: frontend/src/main/pages/HomePage.js (hunk 2 above)
Comment (asharma381) [reply frontend/src/main/pages/HomePage.js:32]: removed this

File: frontend/src/main/components/BasicCourseSearch/BasicCourseSearchForm.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	
+	const firstDepartment = allTheSubjects[0].subjectCode;
+    // Stryker disable next-line all : not sure how to test/mock local storage
+	const [quarter, setQuarter] = useState(localQuarter || quarters[0].yyyyq);
+    // Stryker disable next-line all : not sure how to test/mock local storage
+	const [subject, setSubject] = useState(localSubject || firstDepartment);
+    // Stryker disable next-line all : not sure how to test/mock local storage
+	const [level, setLevel] = useState(localLevel || "U");
+    // Stryker disable next-line all : not sure how to test/mock local storage
+	const [errorNotified, setErrorNotified] = useState(false);
```
Comment (LucienLuc) [inline frontend/src/main/components/BasicCourseSearch/BasicCourseSearchForm.js:32]: You can make this cleaner with 
```
// Stryker disable all
// Code you want Stryker to ignore...
// Stryker restore all
```

File: frontend/src/main/components/BasicCourseSearch/BasicCourseSearchForm.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+			// 		appearance: "error",
+			// 	});
+			// }
+			setCourseJSON(courseJSON);
+		});
+	};
+
+	const handleLevelOnChange = (level) => {
+		// Stryker disable next-line all : not sure how to test/mock local storage
+        localStorage.setItem("BasicSearch.CourseLevel", level);
```
[the remark this answers is not in the document]
Comment (LucienLuc) [reply frontend/src/main/components/BasicCourseSearch/BasicCourseSearchForm.js:47]: Don't think so, you need to pass the setLevel to the component props for it to update.

[the remark this answers is not in the document]
Comment (LucienLuc) [reply frontend/src/main/components/BasicCourseSearch/BasicCourseSearchForm.js:43]: After discussing with Luke, yea local storage is set in components so we shouldn't need them in form

**Review by nazryb — COMMENTED** [review]
lgtm

**Review by LucienLuc — APPROVED** [review]
LGTM!

**Review by nazryb — APPROVED** [review]

File: frontend/src/fixtures/ucsbSubjectsFixtures.js [hunk 5]
```diff
@@ -1,39 +0,0 @@
-const ucsbSubjectsFixtures = {
```
Comment (pconrad) [inline frontend/src/fixtures/ucsbSubjectsFixtures.js:1]: Why are we deleting this file?

File: frontend/src/main/components/UCSBSubjects/UCSBSubjectsTable.js [hunk 6]
```diff
@@ -1,72 +0,0 @@
-import React from "react";
```
Comment (pconrad) [inline frontend/src/main/components/UCSBSubjects/UCSBSubjectsTable.js:1]: Why are we deleting this file?

File: frontend/src/main/utils/UCSBSubjectUtils.js [hunk 7]
```diff
@@ -1,17 +0,0 @@
-import { toast } from "react-toastify";
-
-export function onDeleteSuccess(message) {
-    console.log(message);
-    toast(message);
-}
```
Comment (pconrad) [inline frontend/src/main/utils/UCSBSubjectUtils.js:6]: Why are we deleting this file?

File: frontend/src/stories/components/UCSBSubjects/UCSBSubjectsTable.stories.js [hunk 8]
```diff
@@ -1,29 +0,0 @@
-import React from 'react';
```
Comment (pconrad) [inline frontend/src/stories/components/UCSBSubjects/UCSBSubjectsTable.stories.js:1]: Why are we deleting this file?

File: frontend/src/tests/components/UCSBSubjects/UCSBSubjectsTable.test.js [hunk 9]
```diff
@@ -1,154 +0,0 @@
-import { fireEvent, render, waitFor } from "@testing-library/react";
-import { ucsbSubjectsFixtures } from "fixtures/ucsbSubjectsFixtures";
-import UCSBSubjectsTable from "main/components/UCSBSubjects/UCSBSubjectsTable"
-import { QueryClient, QueryClientProvider } from "react-query";
-import { MemoryRouter } from "react-router-dom";
-import { currentUserFixtures } from "fixtures/currentUserFixtures";
-
```
Comment (pconrad) [inline frontend/src/tests/components/UCSBSubjects/UCSBSubjectsTable.test.js:7]: Why are we deleting this file?

File: frontend/src/tests/utils/UCSBSubjectUtils.test.js [hunk 10]
```diff
@@ -1,59 +0,0 @@
-import { onDeleteSuccess, cellToAxiosParamsDelete } from "main/utils/UCSBSubjectUtils";
-import mockConsole from "jest-mock-console";
-
-const mockToast = jest.fn();
-jest.mock('react-toastify', () => {
-    const originalModule = jest.requireActual('react-toastify');
-    return {
-        __esModule: true,
```
Comment (pconrad) [inline frontend/src/tests/utils/UCSBSubjectUtils.test.js:8]: Why is this being deleted?

**Review by pconrad — COMMENTED** [review]
I see that several files are being deleted in this PR.

I'm confused as to why they are being deleted?   It looks as if these are files that belong in the project?

**Review by pconrad — CHANGES_REQUESTED** [review]
The 5pm-2 team will be really irritated if you delete the files they worked very hard to add to the project.

Please look at everything you are deleting, and see whether you really should be deleting it.
I think you'll find the answer is: NO you should not be.

File: frontend/src/fixtures/ucsbSubjectsFixtures.js (hunk 5 above)
Comment (sidd328) [reply frontend/src/fixtures/ucsbSubjectsFixtures.js:1]: Accidental commit for another PR was reverted making it look like the file is deleted

**Pull request closed without merging**